### PR TITLE
pluma.test: split test.py in different files

### PR DESCRIPTION
**Pull request by Pixep**
`test.py` was way to big, step toward simplifying the whole architecture

### CODE REVIEW

**Comment by mjftw** [comment]
> `test.py` was way to big, step toward simplifying the whole architecture

Completely agree, all of the test framework files and classes are way too heavy.
I've wanted to split them apart for ages so thanks for this!

File: pluma/test/__init__.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
-from .exceptions import *
-from .test import TestBase, TestCore, TestRunner
-from .unittest import deferred_function
-from .testcontroller import TestController
-from .commandrunner import CommandRunner
-from .shelltest import ShellTest
-from .executabletest import ExecutableTest
+from .exceptions import TestingException, TaskFailed, AbortTesting, \
+    AbortTestingAndReport  # noqa: F401
```
Comment (mjftw) [inline pluma/test/__init__.py:2]: What are all these `# noqa: F401` lines about?

File: pluma/test/testrunner.py [hunk 2]
```diff
@@ -1,233 +1,19 @@
...
-        self.board.log("\n=== POST BOARD OFF ===", color='blue', bold=True)
-
-    def report(self):
-        self.board.log("\n=== REPORT ===", color='blue', bold=True)
+from pluma.test import TestCore, TestBase, TestingException, \
+    AbortTesting, AbortTestingAndReport
 
 
 class TestRunner():
+    """Orchestrates tests globally"""
```
Comment (mjftw) [inline pluma/test/testrunner.py:15]: Might be worth mentioning that It only orchestrates a set of tests to run a single time.
Only half the picture with `TestController`.

**Review by mjftw — CHANGES_REQUESTED** [review]

File: pluma/test/__init__.py (hunk 1 above)
Comment (Pixep) [reply pluma/test/__init__.py]: Yep, ignored it in Codacy

File: pluma/test/testrunner.py (hunk 2 above)
Comment (Pixep) [reply pluma/test/testrunner.py]: Updated

**Review by mjftw — APPROVED** [review]
Looks great 👍🏼

**Pull request merged**